core/src/Kokkos_ExecPolicy.hpp

```diff
@@ -983,7 +983,16 @@ template <typename Rank, typename TeamHandle, typename Lambda,
 KOKKOS_INLINE_FUNCTION void parallel_reduce(
     TeamThreadMDRange<Rank, TeamHandle> const& policy, Lambda const& lambda,
     ReducerValueType& val) {
+  static_assert(/*!Kokkos::is_view_v<ReducerValueType> &&*/
```

Comment:
`is_view_v` requires pulling in `Kokkos_View.hpp` which in turn pulls in this header. To avoid the circular dependency, I just commented this particular check.